src/libutils/oglapphelpers/metalapp.mm

```diff
@@ -0,0 +1,380 @@
+// SPDX-License-Identifier: BSD-3-Clause
```

Comment:
You mean instead of objective-C? We can go for metalcpp bindings, but I feel objective-c is the better choice that gives us more control on what happens. After all, objective-c and swift are the main languages for implementing metal apps. Since this is apple platform specific code, that shouldn't be a problem.

Reply:
Good point.